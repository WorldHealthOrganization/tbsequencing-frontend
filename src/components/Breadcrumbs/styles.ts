import { css } from '@emotion/react';
import appColors from '../../styles/colors';

const breadcrumbsMargin = 42;

const baseStyle = {
  fontSize: 16,
};

export const active = css({
  ...baseStyle,
  color: appColors.secondary.whoBlue,
});

export const inactive = css({
  ...baseStyle,
  color: appColors.neutral.black,
});

export const breadcrumbs = css({
  marginTop: breadcrumbsMargin,
  marginBottom: breadcrumbsMargin,
});
