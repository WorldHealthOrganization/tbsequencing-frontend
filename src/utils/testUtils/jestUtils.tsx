import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
