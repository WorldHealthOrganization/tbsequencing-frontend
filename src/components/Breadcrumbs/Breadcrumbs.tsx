/** @jsxImportSource @emotion/react */
import React from 'react';
import { Breadcrumbs as MUIBreadcrumbs, Link } from '@mui/material';
import * as styles from './styles';
import PrimaryText from '../typography/PrimaryText';

export interface IBreadcrumb {
  label: string;
  href: string;
  isActive: boolean;
}

interface Props {
  breadcrumbs: IBreadcrumb[];
}

export const Breadcrumbs = ({ breadcrumbs }: Props) => (
  <MUIBreadcrumbs css={styles.breadcrumbs}>
    {breadcrumbs.map(({ href, label, isActive }) => {
      const linkStyle = isActive ? styles.active : styles.inactive;

      return (
        <Link
          key={label}
          href={href}
          underline="none"
        >
          <PrimaryText style={linkStyle}>{label}</PrimaryText>
        </Link>
      );
    })}
  </MUIBreadcrumbs>
);

export default Breadcrumbs;
