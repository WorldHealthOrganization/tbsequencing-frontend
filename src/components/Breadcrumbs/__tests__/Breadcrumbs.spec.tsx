import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../utils/testUtils/jestUtils';
import Breadcrumbs, { IBreadcrumb } from '../Breadcrumbs';

describe('Breadcrumbs component test', () => {
  const mockConfig:IBreadcrumb[] = [
    { href: 'mockHref', label: 'mockLabel', isActive: false },
  ];

  test('renders Breadcrumbs component without error', () => {
    render(
      <Breadcrumbs breadcrumbs={mockConfig} />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <Breadcrumbs breadcrumbs={mockConfig} />,
    );

    expect(tree).toMatchSnapshot();
  });
});
