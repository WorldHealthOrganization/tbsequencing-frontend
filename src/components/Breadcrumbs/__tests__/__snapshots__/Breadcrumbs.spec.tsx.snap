// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component test Should render correctly 1`] = `
<nav
  className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1x5o0e1-MuiTypography-root-MuiBreadcrumbs-root-Breadcrumbs"
>
  <ol
    className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      className="MuiBreadcrumbs-li"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1cmksfb-MuiTypography-root-MuiLink-root"
        href="mockHref"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1nrkhfx-MuiTypography-root-BaseText"
        >
          mockLabel
        </p>
      </a>
    </li>
  </ol>
</nav>
`;
